src/include/platform/DeviceInstanceInfoProvider.h

```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/core/CHIPError.h>
+#include <lib/support/Span.h>
+
+namespace chip {
+namespace DeviceLayer {
+
+class DeviceInstanceInfoProvider
+{
+public:
+    DeviceInstanceInfoProvider()          = default;
+    virtual ~DeviceInstanceInfoProvider() = default;
+
+    /**
+     * @brief Obtain the Serial Number from the device's factory data.
+     *
+     * The SerialNumber attribute specifies a human readable serial number
+     *
+     * @param[in,out] aSerialNumber Reference to location where the serial number will be copied
+     * @returns CHIP_NO_ERROR on success, or another CHIP_ERROR from the underlying implementation
+     *          if access fails.
+     */
+    virtual CHIP_ERROR GetSerialNumber(MutableCharSpan & aSerialNumber) = 0;
```

Comment:
nit: I think the `a` prefix in argument names is rather obsolete and exists in some Matter files just because they were copied from other projects. But in general, the convention is not used in new files.

Reply:
I thought that a proper convention was to use `a` prefix. 😅